### DISCUSSION TITLE
Unable to Export .sf2

### DISCUSSION THREAD

**Discussion by Meerkov** [vgmtrans/vgmtrans#268] [Q&A] [answered]
I can't get sounds to play within VGMTrans, and can't export soundfont files.

**To Reproduce**

1. Download the latest build for Windows (89b1fae)
2. Drag and drop any game for the PS1 (for example Vagrant Story)
3. Notice that numerous files appear in the left hand panel
4. Double click any file (Akao Sample, Akao Seq, Akao Instr Set, VGMSampColl, Seq, Etc Etc Etc)
5. Click the Green "Play Button" at the bottom of the window.
6. Nothing Happens
7. Try to export any .seq to MIDI (success)
8. Try to export one of the Instr Sets as a .sf2
9. No file is saved at all... nothing happens...

**Expected behavior**
1. I expect the green play button would work.
2. I expect I'd be able to export .sf2 files

**Comment by sykhro** [comment, marked as answer]
Since MIDI conversion works, you're probably hitting something unsupported with regard to instruments.
We need a better feedback mechanism so that the user understands that if nothing happens, the program is failing to complete one of the conversions

**Comment by Meerkov** [comment]
Any chance there is a work around, or some way to figure out what is failing in the extraction? I'd like to export some music, but so far most tools I've found don't succeed in exporting.

**Comment by sykhro** [comment]
No workarounds, sorry. If it's not supported, nothing will magically figure the format out on its own

**Comment by Meerkov** [comment]
No magic is required. A work around could involve a different tool, or perhaps some kind of pointer to where in the code I should be looking to add some error handling.

I wish people wouldn't jump to the assumption that "magic" is the solution I'm looking for. This happens all the time and it's not helpful.

**Comment by sykhro** [comment]
"sf2 doesn't get exported" isn't something that can be worked around. It just means the particular format for the particular game you're trying to convert is either not implemented or bugged. You said "any game for the PS1", which is vague to say the least, since there's many different sequence formats for PS1 games. If you give me a concrete example, I can tell you where the code for that format is.

**Comment by Meerkov** [comment]
For example Vagrant Story.

**Comment by sykhro** [comment]
Vagrant Story is not fully supported. See #98.

**Reply by Meerkov** [reply]
That doesn't solve the problem (see all the above discussion about how there is no error feedback).

This affects nearly every game I try. Another example is 3Xtreme.

**Reply by sykhro** [reply]
> This affects nearly every game I try. Another example is 3Xtreme.

What do 3Xtreme and Vangrant Story have in common? They're PSX games. This assures nothing about their sound formats.
You're seeing one thing, that is failure to export or detect music data, which can be caused by:
- a bug in the parser
- a bug in the scanner
- game having an unsupported format
- game doing something different despite sharing the same format with other games
- your music dump/game rom being corrupted

Now, to me "program won't show any data because it can't detect them" is obvious. But it might not be the same for everyone, and in this I agree with you that some sort of feedback is required. But this is a different matter.
There's nothing of value in a bug report that says "I can't export files", because we don't know what you're trying to export and we need to know in order to tell you "oh that game uses a custom special format that VGMTrans will never support" (e.g. club penguin). There's nothing of value in saying "for example Vagrant Story", as there's only a limited amount of games that have the same music format as Vagrant Story. Be explicit in describing your issue and maybe a workaround can be found or the exact cause can be identified.

**Reply by Meerkov** [reply]
Hi, thanks for taking a moment to reply. We agree that feedback is missing, and required. While you may disagree with the original report, we agree that there is something of value that this issue is reporting. I'd like to further bolster that point of agreement. This doesn't affect only Vagrant Story, or only PS1 games even. Those are merely examples for reproduction purposes. Here are other examples of people having the same issue (failing to export .sf2 files and not receiving any error messages of something failing):

https://github.com/vgmtrans/vgmtrans/issues/251
https://github.com/vgmtrans/vgmtrans/issues/163
https://github.com/vgmtrans/vgmtrans/issues/195
https://github.com/vgmtrans/vgmtrans/issues/166
https://github.com/vgmtrans/vgmtrans/issues/137
https://github.com/vgmtrans/vgmtrans/issues/59
https://github.com/vgmtrans/vgmtrans/issues/195
... etc

I appreciate that the desire is to jump right to the finish and fix the parsing glitch (or add file support, or whatever the underlying cause turns out to be in particular), but I think there is an intermediate fix to add better error notifications to the UI, logs, etc. That would go a long way to users being able to "help themselves" by immediately understanding if a file is not supported, and why, so that they can produce better bug reports, and developers and contributors can more easily add/fix support for those features.

As such, I've marked your earlier reply that VGM Trans is missing proper notifications as the answer for this problem. I'd say approximately ... 10% of bugs in this project mention this lack of error notifications when exporting files, so this feature would go a long way for the community, I'd say.

**Reply by sykhro** [reply]
I have been thinking about this, I'm not sure what's the best way to deal with it.
A message box could be simple and effective but perhaps a bit intrusive.
Something like GTK's in-app notification would probably do the trick, but there's no such thing in Qt and it's probably not worth the hassle of implementing it. 
![immagine](https://user-images.githubusercontent.com/8491400/150689134-14bf4ca8-9ee2-4335-955e-d724d803bb9a.png)

**Reply by sykhro** [reply]
The code knows when no export/detection happens (involved functions return `false`) so there's not a whole lot to do to wire it in
